airbyte-integrations/connectors/destination-chroma/destination_chroma/indexer.py

```diff
@@ -0,0 +1,107 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import uuid
+from typing import List
+
+import chromadb
+from airbyte_cdk.destinations.vector_db_based.document_processor import METADATA_RECORD_ID_FIELD, METADATA_STREAM_FIELD, Chunk
+from airbyte_cdk.destinations.vector_db_based.indexer import Indexer
+from airbyte_cdk.destinations.vector_db_based.utils import format_exception
+from airbyte_cdk.models import ConfiguredAirbyteCatalog
+from airbyte_cdk.models.airbyte_protocol import DestinationSyncMode
+from chromadb.config import Settings
+from destination_chroma.config import ChromaIndexingConfigModel
+from destination_chroma.utils import is_valid_collection_name
+
+
+class ChromaIndexer(Indexer):
+    def __init__(self, config: ChromaIndexingConfigModel):
+        super().__init__(config)
+        self.collection_name = config.collection_name
+
+    def check(self):
+        collection_name_validation_error = is_valid_collection_name(self.collection_name)
+        if collection_name_validation_error:
+            return collection_name_validation_error
+        
+        auth_method = self.config.auth_method
+        if auth_method.mode == "persistent_client" and not auth_method.path.startswith("/local/"):
+                return "Path must be prefixed with /local"
+        
+        client = self._get_client()
+        try:
+            heartbeat = client.heartbeat()
+            if not heartbeat:
+                return "Chroma client server is not alive"
+            collection = client.get_or_create_collection(name=self.collection_name)
+            count = collection.count()
+            if count != 0 and not count:
+                return f"unable to get or create collection with name {self.collection_name}"
+            return
+        except Exception as e:
+            return format_exception(e)
+        finally:
+            del client
+
+    def index(self, document_chunks: List[Chunk], delete_ids: List[str]) -> None:
+        if len(delete_ids) > 0:
+            self._delete_by_filter(field_name=METADATA_RECORD_ID_FIELD, field_values=delete_ids)
+        entities = []
+        for i in range(len(document_chunks)):
+            chunk = document_chunks[i]
+            entities.append(
+                {"id": str(uuid.uuid4()), "embedding": chunk.embedding, "metadata": chunk.metadata, "document": chunk.page_content}
```

Comment:
If the metadata object contains something that is not a string, a list of strings or a number it will fail to index it. In the Weaviate PR I have the same issue, could you use a similar strategy here? Normalize the metadata per chunk by checking whether its type is supported and if not, serialize it as a JSON string: https://github.com/airbytehq/airbyte/pull/30151/files#diff-605751b53927fd85aee9d14cf7cc427ddc74aad7c986c31bffc1b06e528a5923R111